src/dtx/dtx_common.c

```diff
@@ -508,7 +508,7 @@ dtx_conflict(daos_handle_t coh, struct dtx_leader_handle *dlh, uuid_t po_uuid,
 	if (abort_dtes != NULL)
 		D_FREE(abort_dtes);
 
-	return rc > 0 ? 0 : rc;
+	return rc > 0 ? -DER_INVAL : rc;
```

Comment:
seems no case above can get positive rc?

Reply:
I have checked the logic, we do not return positive value.

Reply:
then why don't we assert? It means a logic bug in our code and we want to detect it.